File: sdk/typescript/test/fixture-provider.js
#!/usr/bin/env node
import { completeResult, createProvider, serveProvider } from "../src/index.js";

const provider = createProvider({
  name: "typescript-fixture",
  version: "1.0.0",
  tools: [{
    name: "fixture.echo",
    description: "Echo a message while a dependency writes a diagnostic.",
    inputSchema: {
      type: "object",
      properties: { message: { type: "string", minLength: 1 } },
      required: ["message"],
      additionalProperties: false,
    },
    outputSchema: { type: "object" },
    effect: "read",
    handler: ({ message }) => {
      console.log("simulated third-party diagnostic");
      if (typeof message !== "string") throw new Error("message is required");
      return completeResult({ structuredContent: { message } });
    },
  }],
});

process.exitCode = await serveProvider(provider);

File: sdk/python/tests/fixture_provider.py
#!/usr/bin/env python3
from maelys_mcp_provider import Tool, create_provider, serve_provider


def echo(arguments):
    print("simulated third-party diagnostic")
    if not isinstance(arguments.get("message"), str):
        raise ValueError("message is required")
    return {"message": arguments["message"]}


provider = create_provider("python-fixture", "1.0.0", [Tool(
    name="fixture.echo",
    description="Echo a message while a dependency writes a diagnostic.",
    input_schema={
        "type": "object",
        "properties": {"message": {"type": "string", "minLength": 1}},
        "required": ["message"],
        "additionalProperties": False,
    },
    output_schema={"type": "object"},
    effect="read",
    handler=echo,
)])

raise SystemExit(serve_provider(provider))
